src/renderers/dom/client/ReactBrowserEventEmitter.js

```diff
@@ -328,6 +369,14 @@ var ReactBrowserEventEmitter = Object.assign({}, ReactEventEmitterMixin, {
         isListening[dependency] = true;
       }
     }
+
+    return function() {
```

Comment:
@syranide 

This function can accept documents and other types of targets. In both these cases it returns a remover that removes all registered listeners. To give a consistent interface it must therefore always return a remover for all types of target.

In this case, the caller knows if its adding to a document, and it does the right thing with the returned remover.

Its simply trying to match the style that whoever registers a listener also takes care of removing them at the right time.